client/my-sites/theme/main.jsx

```diff
@@ -554,8 +573,8 @@ export default connect(
 		const backPath = getBackPath( state );
 		const currentUserId = getCurrentUserId( state );
 		const isCurrentUserPaid = isUserPaid( state, currentUserId );
-		const themeDetails = getThemeDetails( state, id );
-
+		const theme = selectedSite ? getTheme( state, selectedSite.ID, id ) : getTheme( state, 'wpcom', id );
```

Comment:
Oh, this is another possible (quite likely) culprit for why single-site mode doesn't work.

Keep in mind we only store theme data for any wpcom site in the `wpcom` subtree. This thing however will look in a subtree named for the `siteId`. We only do that for Jetpack sites.